packages/contracts-bedrock/src/L2/L1BlockInterop.sol

```diff
@@ -2,108 +2,103 @@
-            if xor(add(165, mul(dependencySetSize_, 0x20)), calldatasize()) {
-                mstore(0x00, 0x44165b6a) // 0x44165b6a is the 4-byte selector of "DependencySetSizeMismatch()"
-                revert(0x1C, 0x04) // returns the stored 4-byte selector from above
-            }
-
-            // Use memory to hash and get the start index of dependencySet
-            mstore(0x00, dependencySet.slot)
-            let dependencySetStartIndex := keccak256(0x00, 0x20)
-
-            // Iterate over calldata dependencySet and write to store dependencySet
-            for { let i := 0 } lt(i, dependencySetSize_) { i := add(i, 1) } {
-                // Load value from calldata and write to storage (dependencySet) at index
-                let val := calldataload(add(165, mul(i, 0x20)))
-                sstore(add(dependencySetStartIndex, i), val)
-            }
-
-            // Update length of dependencySet array
-            sstore(dependencySet.slot, dependencySetSize_)
-        }
-    }
-
     /// @notice Returns true if a chain ID is in the interop dependency set and false otherwise.
-    ///         Every chain ID is in the interop dependency set of itself.
+    ///         The chain's chain ID is always considered to be in the dependency set.
     /// @param _chainId The chain ID to check.
     /// @return True if the chain ID to check is in the interop dependency set. False otherwise.
     function isInDependencySet(uint256 _chainId) public view returns (bool) {
-        // Every chain ID is in the interop dependency set of itself.
-        if (_chainId == block.chainid) {
-            return true;
-        }
-
-        uint256 length = dependencySet.length;
-        for (uint256 i = 0; i < length;) {
-            if (dependencySet[i] == _chainId) {
-                return true;
-            }
-            unchecked {
-                i++;
-            }
-        }
-
-        return false;
+        return _chainId == block.chainid || dependencySet.contains(_chainId);
     }
 
     /// @notice Returns the size of the interop dependency set.
     /// @return The size of the interop dependency set.
     function dependencySetSize() external view returns (uint8) {
-        return uint8(dependencySet.length);
+        return uint8(dependencySet.length());
+    }
+
+    /// @notice Sets static configuration options for the L2 system. Can only be called by the special
+    ///         depositor account.
+    /// @param _type  The type of configuration to set.
+    /// @param _value The encoded value with which to set the configuration.
+    function setConfig(ConfigType _type, bytes calldata _value) external {
+        if (msg.sender != DEPOSITOR_ACCOUNT()) revert NotDepositor();
+
+        // For GAS_PAYING_TOKEN config type
+        if (_type == ConfigType.GAS_PAYING_TOKEN) {
+            (address token, uint8 decimals, bytes32 name, bytes32 symbol) =
+                abi.decode(_value, (address, uint8, bytes32, bytes32));
+
```

Comment:
I recommend using an encoding/decoding library that wraps `abi.decode`, this would be the same library used to `abi.encode`, two functions for each `ConfigType`, an `encode` and a `decode`